src/test/java/com/facebook/openwifirrm/optimizers/tpc/RandomTxPowerInitializerTest.java

```diff
@@ -48,14 +49,36 @@ void test1() throws Exception {
 		dataModel.latestState.put(deviceA, createState());
 		dataModel.latestState.put(deviceB, createState());
 
-		final int txPower = 16;
+		// Test providing tx power
+		for (int txPower: Arrays.asList(-1, 16, 27)) {
+			TPC optimizer = new RandomTxPowerInitializer(
+					dataModel, TEST_ZONE, deviceDataManager, txPower
+			);
+			Map<String, Map<String, Integer>> txPowerMap =
+					optimizer.computeTxPowerMap();
+
+			for (String band : UCentralConstants.BANDS) {
+				assertEquals(txPower, txPowerMap.get(deviceA).get(band));
+				assertEquals(txPower, txPowerMap.get(deviceB).get(band));
+			}
+		}
+
+		// Test default tx power
 		TPC optimizer = new RandomTxPowerInitializer(
-			dataModel, TEST_ZONE, deviceDataManager, txPower
+				dataModel, TEST_ZONE, deviceDataManager
 		);
 		Map<String, Map<String, Integer>> txPowerMap =
-			optimizer.computeTxPowerMap();
+				optimizer.computeTxPowerMap();
 
-		assertEquals(txPower, txPowerMap.get(deviceA).get(UCentralConstants.BAND_5G));
-		assertEquals(txPower, txPowerMap.get(deviceB).get(UCentralConstants.BAND_5G));
+		for (String band : UCentralConstants.BANDS) {
+			assertEquals(
+					RandomTxPowerInitializer.DEFAULT_TX_POWER,
+					txPowerMap.get(deviceA).get(band)
+			);
+			assertEquals(
+					RandomTxPowerInitializer.DEFAULT_TX_POWER,
+					txPowerMap.get(deviceB).get(band)
+			);
+		}
```

Comment:
er... @pohanhf why does this class not actually assign a random tx power (hardcoding default=23)??

Reply:
Oh no... this function was "single tx power assignment" instead of random tx power assignment. We changed the name wrongly before importing the code from internal to external. @wuweicaifb, could you please help make this function right by actually randomly assigning tx power for the devices in this PR?  cc. @elludraon

Reply:
Do we have a random range for the tx power?

Reply:
[0, 30]
https://ucentral.io/docs/ucentral-schema.html#radios_items_tx-power

Reply:
Another question about random here: should we only call random once, so that this Random TPC set that random number to all the bands and devices, or we call random every time for each different bands and devices?

Reply:
Just make all the devices on a band be the same random tx power for now. We will add some functionality to pass arguments to enable the random initializer that assigns each devices on different bands a different tx power along with the random channel initializer.